packages/vaadin-grid/test/column.test.js

```diff
@@ -485,98 +485,66 @@ describe('column', () => {
-      const column = document.createElement('vaadin-grid-column');
-      const template = document.createElement('template');
-      template.classList.add('header');
+    let grid, column;
 
-      column.appendChild(template);
-      column._templateObserver.flush();
+    beforeEach(() => {
+      grid = fixtureSync(`
+        <vaadin-grid>
+          <vaadin-grid-column></vaadin-grid-column>
+        </vaadin-grid>
+      `);
 
-      expect(column._headerTemplate).to.eql(template);
-    });
-
-    it('should pickup footer template', () => {
-      const column = document.createElement('vaadin-grid-column');
-      const template = document.createElement('template');
-      template.classList.add('footer');
-
-      column.appendChild(template);
-      column._templateObserver.flush();
-
-      expect(column._footerTemplate).to.eql(template);
-    });
-
-    it('should pickup body template', () => {
-      const column = document.createElement('vaadin-grid-column');
-      const template = document.createElement('template');
-
-      column.appendChild(template);
-      column._templateObserver.flush();
-
-      expect(column._bodyTemplate).to.eql(template);
-    });
+      grid.items = ['item1', 'item2'];
+      column = grid.firstElementChild;
 
-    it('should re-pickup header template', () => {
-      const column = document.createElement('vaadin-grid-column');
-      const template = document.createElement('template');
-      template.classList.add('header');
-      const template2 = document.createElement('template');
-      template2.classList.add('header');
-
-      column.appendChild(template);
-      column._templateObserver.flush();
-      column.removeChild(template);
-      column.appendChild(template2);
-      column._templateObserver.flush();
-      expect(column._headerTemplate).to.eql(template2);
+      flushGrid(grid);
     });
 
-    it('should pickup new body template', () => {
-      const column = document.createElement('vaadin-grid-column');
-      const template = document.createElement('template');
-      const template2 = document.createElement('template');
+    ['header', 'body', 'footer'].forEach((templateName) => {
+      let cell;
 
-      column.appendChild(template);
-      column._templateObserver.flush();
-      column.removeChild(template);
-      column.appendChild(template2);
-      column._templateObserver.flush();
-
-      expect(column._bodyTemplate).to.eql(template2);
-    });
-
-    it('should prepare template when added lazily', () => {
-      const column = document.createElement('vaadin-grid-column');
-      const template = document.createElement('template');
+      beforeEach(() => {
+        if (templateName === 'header') {
+          cell = grid.$.header.children[0];
+        }
+        if (templateName === 'footer') {
+          cell = grid.$.footer.children[0];
+        }
+        if (templateName === 'body') {
+          cell = grid.$.items.children[0];
```

Comment:
This variable should be `row`, since `grid.$.items.children` are `<tr>` elements

Reply:
Thanks! Fixed.